### PR TITLE
Decode RequestError body

### PR DESCRIPTION
**Pull request by djones6**
Replacement PR for #18 targeting the `develop` branch.

Add support for populating/decoding the body of a RequestError.

Requires: IBM-Swift/SwiftyRequest#18 (SwiftyRequest 1.1.2) and IBM-Swift/Kitura#1214 (Kitura 2.3).

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -30,10 +30,10 @@ let package = Package(
     ],
     dependencies: [
         // Dependencies declare other packages that this package depends on.
-        .package(url: "https://github.com/IBM-Swift/SwiftyRequest.git", .upToNextMinor(from: "1.1.0")),
-        .package(url: "https://github.com/IBM-Swift/KituraContracts.git", .upToNextMinor(from: "0.0.18")),
+        .package(url: "https://github.com/IBM-Swift/SwiftyRequest.git", from: "1.1.0"),
+        .package(url: "https://github.com/IBM-Swift/KituraContracts.git", from: "0.0.19"),
```
Comment (EnriqueL8) [inline Package.swift]: KituraContracts latest version is `0.0.22`

**Comment by djones6** [comment]
@ianpartridge You've already reviewed this under #18 - this requires a new minor of Kitura (2.3) for the tests to pass, and I've taken the opportunity to also remove the minor constraints and update to Swift 4.1.

We should also think about what version this should be tagged as.

**Comment by djones6** [comment]
@ddunn2 can you confirm whether we need to specify the linux `dist` in the `.travis.yml` for the additional jobs?

File: Tests/KituraKitTests/MainTests.swift [hunk 2]
```diff
@@ -390,4 +397,130 @@ class MainTests: XCTestCase {
...
+            XCTAssertNil(users, "Unexpected success value given: \(String(describing: users))")
+            if let error = error {
+                XCTAssertEqual(RequestError.conflict.rawValue, error.rawValue)
+                let body = error.bodyAs(Status.self)
+                XCTAssertNotNil(body, "No body or body is not a Status")
+                if let conflict = body {
+                    XCTAssertEqual(Status("Boo"), conflict)
+                }
+            }
+        expectation1.fulfill()
```
Comment (ianpartridge) [inline Tests/KituraKitTests/MainTests.swift]: Indentation. (same in other tests below)

**Pull request merged**